features/landscape.feature

```diff
@@ -99,21 +58,33 @@ Feature: Enable landscape on Ubuntu
         """
         landscape +yes +enabled
         """
-        When I run `sudo pro disable landscape` with sudo
-        # TODO: Known status bug around not uninstalling landscape: #2840
-        When I run `apt-get remove -y landscape-client` with sudo
+
+        # stopping the service effectively disables it
+        When I run `systemctl stop landscape-client` with sudo
```

Comment:
It wasn't specifically discussed. What do you think makes sense? It is essentially disabled until reboot, so I don't think it is wrong as-is, but perhaps missing information.

Is it common to `stop` the service in practice? I only did it in this test to exercise the scenario.

We could potentially check both `is-active` and `is-enabled` and give a warning if enabled but not active, but I don't know if the complexity is worth it - thoughts?

Reply:
I wasn't suggesting this was wrong, just wanted to know the rationale, and I think the rationale is fine.

It's much more common anyways to restart or outright disable the service rather than stop it. The behaviour as-is is fine.

Reply:
Testing the package from the PPA listed above it seems fine. It refused to disable landscape when asked, and did not disable the `landscape-client` service, through several different `pro detach` / `pro attach` cycles. I'd say this looks fine, though it seems like landscape has been removed from the `pro status` output at this time. I guess I'm not sure if that is intended or not. If this is intended then this looks fine to me.

This should definitely resolve the issue we've seen in support cases, where a client is re-attaching with a new token, and the landscape-client gets disabled. Thankfully, we have only gotten a few cases where clients had to be told to re-name the disabled `client.conf` file and restart the `landscape-client` manually to resolve the disable action.

Reply:
@mdscunningham thank you for taking a look and testing it out!

Currently, landscape should not show up in `pro status` except for on mantic.

After landscape-client is SRU'd to all supported releases, we will start supporting pro enable/disable/status for landscape on older releases too. At that time, `pro detach` **will** disable landscape on e.g. jammy, but after re-attaching, a simple `pro enable landscape --assume-yes` will re-enable landscape with the same settings it had previously (so no more `mv /etc/landscape/client.conf{,.pro-disable-backup}` weirdness).

Does that sound okay?